compiler/compiler.py

```diff
@@ -32,47 +33,38 @@ def compile_upload(upload_id: str, output_format: str='pdf',
     output_format: str
         The desired output format. Default: "pdf". Other potential values:
         "dvi", "html", "ps"
-    fms_endpoint: str
-        The API endpoint for the FileManagementService.
     preferred_compiler: str
         The preferred tex compiler for use with the source package.
-    output_endpoint: str
-        The API endpoint for uploading compiled files.
     """
-
-    if fms_endpoint is None:
-        fms_endpoint = current_app.config['FMS_ENDPOINT']
-
-    if output_endpoint is None:
-        output_endpoint = current_app.config['OUTPUT_ENDPOINT']
-
-
-    # 1. Retrieve the source package
-    fms = FileManagementService(fms_endpoint)
-
-    body, headers = fms.get_upload_content(upload_id)
-    etag = headers['ETag']
+    source_package = filemanager.get_upload_content(upload_id)
 
     # 2. Generate the compiled files
     # 3. Upload the results to output_endpoint
     with TemporaryDirectory(prefix='arxiv') as source_dir,\
          TemporaryDirectory(prefix='arxiv') as output_dir:
-        with tarfile.open(fileobj=body, mode='r:gz') as tar: # type: ignore
+        with tarfile.open(fileobj=source_package.stream, mode='r:gz') as tar: # type: ignore
             tar.extractall(path=source_dir)
 
         # 2. Generate the compiled files
         compile_source(source_dir, output_dir)
 
         # 3. Upload the results to output_endpoint
-        # TODO: upload the output somewhere(?)
+        # 3a. gather the candidate products based on output_format
+        products = [f for f in os.listdir(output_dir) 
+                        if f.endswith(output_format)]
+
+        if len(products) != 1:
+            raise RuntimeWarning("Multiple compilation products")
+
```

Comment:
Somewhere in here is where we would instantiate the [``CompilationProduct``](https://github.com/cul-it/arxiv-compiler/blob/master/compiler/domain.py#L63), which is just a bytesio, checksum, and a ``CompilationStatus`` object. That's all that the storage service (see below) requires.